src/Domain/Entity/PurchaseEntity.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Domain.common;
+
+namespace Domain.Entity;
+
+public class PurchaseEntity : IEntity<Guid>
+{
+    public Guid Id { get; set; }
+
+    [Required(ErrorMessage = ErrorMessages.PurchaseEntityBillingPartyRequired)]
+    public BillingPartyEntity BillingParty { get; set; }
+
+    private ICollection<PurchaseLineItem> _purchases;
+
+    [Required(ErrorMessage = ErrorMessages.PurchaseEntityPurchaseLinesRequired)]
+    public ICollection<PurchaseLineItem> Purchases
+    {
+        get => _purchases;
+        set
+        {
+            ValidatePurchases(value);
+            _purchases = value;
+        }
+    }
+
+    private double? _vatAmount;
+
+    [Range(0, double.MaxValue, ErrorMessage = ErrorMessages.PurchaseEntityVatAmountPositive)]
+    public double? VatAmount
+    {
+        get => _vatAmount;
+        set
+        {
+            ValidateVatAmount(value);
+            _vatAmount = value;
+        }
+    }
+
+    private double? _transportFee;
+    [Range(0, double.MaxValue, ErrorMessage = ErrorMessages.PurchaseEntityTransportFeePositive)]
+    public double? TransportFee
+    {
+        get => _transportFee;
+        set
+        {
+            ValidateTransportFee(value);
+            _transportFee = value;
+        }
+    }
+    
+    private double? _paidAmount;
+    [Range(0, double.MaxValue, ErrorMessage = ErrorMessages.PurchaseEntityPaidAmountPositive)]
+    public double? PaidAmount
+    {
+        get => _paidAmount;
+        set
+        {
+            ValidatePaidAmount(value);
+            _paidAmount = value;
+        }
+    }
+    
+    private string? _remarks;
+    [MaxLength(500, ErrorMessage = ErrorMessages.PurchaseEntityRemarksTooLong)]
+    public string? Remarks
+    {
+        get => _remarks;
+        set
+        {
+            ValidateRemarks(value);
+            _remarks = value;
+        }
+    }
+
+    private void ValidateRemarks(string? value)
+    {
+        if (value != null && value.Length > 500)
+        {
+            throw new DomainValidationException("Remarks", ErrorCode.BadRequest,
+                ErrorMessages.PurchaseEntityRemarksTooLong);
+        }
+    }
```

Comment:
Please dont consider me being too nifty, but null checks are advised to be done as value is null // value is not null